app/controllers/editor.js

```diff
@@ -374,6 +383,14 @@ export default Controller.extend({
 
             return post;
         } catch (error) {
+            // trigger upgrade modal if forbidden error
+            if (error && isForbiddenError(error) && error.payload && error.payload.errors && error.payload.errors[0].message) {
```

Comment:
@kevinansfield I am not 💯 if this is the right place to handle the `403` and trigger the upgrade popup, as the other place I was thinking it might be better is at the top level in application where ajax error handling is done [here](https://github.com/TryGhost/Ghost-Admin/blob/master/app/routes/application.js#L128) instead. 

The idea about putting in application was that it becomes app level generic handling to throw the popup whenever `403-HostLimitError` comes, but was not sure how that will fit with controller/template. Will be good interested in hearing your thoughts on how to approach this :)

Reply:
No need to do anything generic at all at the moment, let's not pre-optimise until we have other use cases :) Explicit handling in the editor controller's save task is 👌